agent/util-scripts/pbench-tool-meister-stop

```diff
@@ -45,8 +49,34 @@ def wait_for_pid(pid):
         time.sleep(0.1)
 
 
+class RedisServer(RedisServerCommon):
+    """RedisServer - an encapsulation of the handling of the Redis server
+    specification for pbench-tool-meister-stop.
+
+    The constructor is enhanced to find the optional local Redis server pid
+    file, and the additional method, locally_managed(), keys off of its
+    presence.
+    """
+
+    def __init__(self, spec: str, benchmark_run_dir: Path, def_host_name: str):
+        super().__init__(spec, def_host_name)
+        try:
+            self.pid_file = (
+                benchmark_run_dir / "tm" / f"redis_{self.def_port:d}.pid"
+            ).resolve(strict=True)
+        except FileNotFoundError:
+            pass
+
+    def locally_managed(self):
+        return self.pid_file is not None
+
+
 def graceful_shutdown(
```

Comment:
There's no description (nor type hint) for the return value from this `graceful_shutdown()` (nor for `locally_managed()`, but that one is easier to infer).